src/dmd/dtemplate.d

```diff
@@ -5808,6 +5803,36 @@ extern (C++) class TemplateInstance : ScopeDsymbol
     TemplateInstance tnext;     // non-first instantiated instances
     Module minst;               // the top module that instantiated this instance
 
+    uint inuse;                  // for recursive expansion detection
+    ushort nest;                 // for recursive pretty printing detection
```

Comment:
I'm not sure whether this is right, please lmk.

Reply:
globals.d define the max value to 500 so ushort is okay.